tools/sklearn/utils.py

```diff
@@ -55,6 +54,7 @@ def find_class(self, module, name):
                 # TODO: replace with a whitelist checker
                 if fullname not in sk_whitelist['SK_NAMES'] + sk_whitelist['SKR_NAMES'] + sk_whitelist['XGB_NAMES'] + sk_whitelist['NUMPY_NAMES'] + good_names:
                     print("Warning: global %s is not in pickler whitelist yet and will loss support soon. Contact tool author or leave a message at github.com" % fullname)
+                __import__(module, level=0)
```

Comment:
This could damp the whitelist checker. I would avoid this and add needed import in the header.